internal/pkg/template/diff/diff.go

```diff
@@ -87,7 +87,7 @@ type seqItemNode struct {
 type From []byte
 
 // Parse constructs a diff tree that represent the differences of a YAML document against the From document.
-func (from From) Parse(to []byte) (Tree, error) {
+func (from From) Parse(to []byte, overrider Overrider) (Tree, error) {
```

Comment:
Actually, thinking some more maybe we can start by being super specific rather than the more generic Overrider term:

```go
func (from From) Parse(to []byte) (Tree, error)
func (from From) ParseWithIgnorer(to []byte, ignorer Ignorer) (Tree, error)
```
and clients can write

```go
from.ParseWithIgnorer(to, diff.IgnoreCFNMetadata())
```

Reply:
I ended up just did to be super duper specific

```
func (from From) Parse(to []byte) (Tree, error)
func (from From) ParseWithCFNIgnorer(to []byte) (Tree, error)
```

and not exposing anything from `override` - not the interface `overrider` nor any of the specific overrider.